added moon location as black dot

rdoshi99 opened the pull request:
- added function get_moonloc which returns a SkyCoord object  to retrieve the RA and DEC for the moon
- plotted the location as a black dot on the skypath plot
- the coordinates seem a little off
- still working on the phase of the moon
- in progress issue #27

rdoshi99 committed:
added moon location as black dot

sbailey commented on py/surveyqa/nightly.py:
```diff
@@ -194,10 +211,16 @@ def get_skypathplot(exposures, tiles, width=600, height=300):
     fig.asterisk(first['RA'], first['DEC'], size=10, line_width=1.5, fill_color=None, color='gold')
     
+    #- Adds moon location at midnight on NIGHT
+    mjd_midnight = int(exposures['MJD'][0])
```
int(MJD) is midnight in Greenwich, England, instead of midnight at Kitt Peak in Arizona (7 hours behind GMT/UT/UTC).  `int(exposures['MJD'][0])` is also fragile since it depends upon whether the first exposure of the night was before or after 5pm local time (i.e. before or after midnight UT).  This might be the source of your "the coordinates seem a little off", depending upon what you were comparing against.

Consider parsing the "NIGHT" column (YEARMMDD of the sunset) into the MJD of local midnight instead.

Check the case of night 20191215 by hand.  It looks like our bright time exposure path got very close to the moon, which shouldn't happen (it could be a bug in the scheduler, or in the moon location calculation).  Compare with the `MOONSEP` and `MOONFRAC` values for that night (which also could have bugs, so approach with some skepticism and let's track this down).

sbailey commented on py/surveyqa/nightly.py:
```diff
@@ -151,6 +153,21 @@ def get_nightlytable(exposures):
     return nightly_table
 
 
+def get_moonloc(mjd_midnight):
```
Detail: although you intend to use this function by calling it with the MJD of a particular midnight, it actually works to return the moon location at any MJD whether or not it was midnight.  i.e. it would be better for this function to be just `get_moonloc(mjd)` (valid for any MJD) or `get_moonloc(night)` (forcing the answer to be for midnight on a particular night=YEARMMDD of sunset).

sbailey commented on py/surveyqa/nightly.py:
```diff
@@ -194,10 +211,16 @@ def get_skypathplot(exposures, tiles, width=600, height=300):
+    mjd_midnight = int(exposures['MJD'][0])
+    moon_loc = get_moonloc(mjd_midnight)
+    ra, dec = float(moon_loc.ra.to_string(decimal=True)), float(moon_loc.dec.to_string(decimal=True))
+    fig.circle(ra, dec, size=10, color='black')
```
Minor: let's make the moon color "gold" and pick a different color for the start-of-night star symbol.

sbailey reviewed (changes requested):
A few comments/requests inline.  Minor merge conflicts have arisen which could be resolved by the github web editor followed by "git pull" on your local copy.

rdoshi99 committed:
fixed get_moonloc function to parse night instead of the greenwich based MJD

rdoshi99 committed:
Merge branch 'master' into skypathMoon

rdoshi99 commented:
- changed moonloc function to parse the night instead
- changed moon color to gold and orange for the start-of-night star
- checked the case of 20191215 by hand, found that for the tiles near the moon, the moon separation angle is only 5 degrees

sbailey commented on py/surveyqa/nightly.py:
```diff
@@ -151,6 +151,28 @@ def get_nightlytable(exposures):
+    """
+    #- Re-formats night into YYYY-MM-DD HH:MM:SS 
+    iso_format = night[:4] + '-' + night[4:6] + '-' + night[6:] + ' 00:00:00'
+    t_midnight = Time(iso_format, format='iso') + 24*u.hour
```
Almost.  This is midnight GMT, but we want midnight in Arizona after the sunset on night YEARMMDD.  Midnight in AZ is 7am GMT on the next day, so you are correct about the `+ 24*u.hour`, but we need the time portion of the string to be `07:00:00` not `00:00:00`.  Double check me on that...

rdoshi99 committed:
Update nightly.py

Huh. The timezone change is there on my commit and push history, but it is not showing up in the merge conflict.
I changed it directly in the web editor, it should be set now.
